tests/test_spatial_crop.py

```diff
@@ -25,7 +25,7 @@
     [{"roi_start": [0, 0], "roi_end": [2, 2]}, (3, 3, 3, 3), (3, 2, 2, 3)],
     [{"roi_start": [0, 0, 0, 0, 0], "roi_end": [2, 2, 2, 2, 2]}, (3, 3, 3, 3), (3, 2, 2, 2)],
     [{"roi_start": [0, 0, 0, 0, 0], "roi_end": [8, 8, 8, 2, 2]}, (3, 3, 3, 3), (3, 3, 3, 3)],
-    [{"roi_start": [1, 0, 0], "roi_end": [1, 8, 8]}, (3, 3, 3, 3), (3, 0, 3, 3)],
+    # [{"roi_start": [1, 0, 0], "roi_end": [1, 8, 8]}, (3, 3, 3, 3), (3, 0, 3, 3)], # not support during lazy
```

Comment:
When output data contains zero dimension(None), it's not supported in the lazy mode. Even after changing the `func` with `x>=0`, we can get the correct `spatial_size` [here](https://github.com/Project-MONAI/MONAI/blob/26f100aaf27cc9e4fd404d6ec8271b2c5e2ad8f2/monai/transforms/spatial/functional.py#L98), but zero is still not supported by `nn.functional.affine_grid`.
https://github.com/Project-MONAI/MONAI/blob/f519699f4b709aa69f23c252842e4798cac293d8/monai/utils/misc.py#L195
https://github.com/Project-MONAI/MONAI/blob/26f100aaf27cc9e4fd404d6ec8271b2c5e2ad8f2/monai/transforms/spatial/functional.py#L98

Reply:
sure, that's expected, I think that's an ill-pose case, raising an error is fine..

Reply:
ok, will add it.

Reply:
updated in this [commit](https://github.com/Project-MONAI/MONAI/pull/6068/commits/4a7b4bac7185ad6709f7f71bdfa52378191589cf).